spec/lib/apipie/generator/swagger/param_description/builder_spec.rb

```diff
@@ -102,23 +102,49 @@
 
     subject { generated_block }
 
-    context 'when is not required' do
-      let(:base_param_description_options) { { required: false } }
+    context 'required: true' do
+      let(:base_param_description_options) { { required: true } }
+
+      it 'does not output an option without default warning' do
+        expect { subject }.not_to output(
+          /is optional but default value is not specified/
+        ).to_stderr
+      end
+    end
 
-      context 'and no default is given' do
-        before { param_description_options.delete(:default) }
+    context 'required: false' do
+      context 'no warning if default_value: nil' do
+        let(:base_param_description_options) do
+          { required: false, default_value: nil }
+        end
 
-        it 'outputs an option without default warning' do
-          expect { subject }.to output(/is optional but default value is not specified/).to_stderr
+        it 'will not warn' do
+          expect { subject }.not_to output(
+            /is optional but default value is not specified/
+          ).to_stderr
         end
       end
-    end
 
-    context 'when is required' do
-      let(:base_param_description_options) { { required: true } }
+      context 'no warning if default_value: 123' do
```

Comment:
NIT We can use the `context` to describe and the environment and let the `it` hint what the result should be  
```suggestion
      context 'when default_value is 123' do
```